File: LeetCode/121. Best Time to Buy and Sell Stock.cpp
/***  KNOWLEDGE IS POWER  ***/

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int l = INT_MAX , ans = 0 , p = 0 ; 
        for(int i = 0; i < prices.size(); i++)
        {
            if(prices[i] < l)
            {
                l = prices[i];
            }
            p = prices[i] - l ;
            if(ans < p)
            {
                ans = p ;
            }
        }
        return ans ;
    }
};

File: LeetCode/121. Best Time to Buy and Sell Stock.py
###  KNOWLEDGE IS POWER  ###

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = 10000 
        ans = 0 
        p = 0
        for i in prices:
            if i < l:
                l = i
            p = i - l 
            if ans < p:
                ans = p 
        return ans 
